File: tests/cypress/cypress/elements/pages/logs/throttle.rb
require_relative './logs.rb'

class ThrottleLog < Logs
  set_url_matcher /logs\/throttle/

  initialize
    @menu_item = 'Throttling'
  }

  generate_data(
    count: 250,
    ip_address: nil,
    timestamp_min: nil,
    timestamp_max: nil,
    hits: nil,
    locked_out: nil
    )
    command = "cd fixtures && php throttlingLog.php"

    if count
      command += " --count " + count.to_s
    }

    if ip_address
      command += " --ip-address '" + ip_address.to_s + "'"
    }

    if timestamp_min
      command += " --timestamp-min " + timestamp_min.to_s
    }

    if timestamp_max
      command += " --timestamp-max " + timestamp_max.to_s
    }

    if hits
      command += " --hits " + hits.to_s
    }

    if locked_out
      command += " --locked-out"
    }

    command += " > /dev/null 2>&1"

    system(command)
  }
}

File: tests/cypress/cypress/elements/pages/logs/throttle.js
// import ControlPanel from '../ControlPanel'
// import Logs from './logs.js'
// class ThrottleLog extends ControlPanel {
//   constructor() {
//     super()

//     this.urlMatcher = 'admin.php?/cp/logs/throttle';
//     this.count = 250;
//     this.ip_address = null;
//     this.timestamp_min = null;
//     this.timestamp_max = null;
//     this.hits = null;
//     this.locked_out = null;
//   }

//   runner(){
//     let command = `cd support/fixtures && php throttlingLog.php `;
//     if(this.count > 0){
//       command += ` --count ` + this.count.toString();
//     }

//     if(this.ip_address != null){
//       command += `--ip-address ` + this.ip_address.toString();
//     }

//     if(this.timestamp_min != null){
//       command += ` --timestamp-min ` + timestamp_min.toString()
//     }
//     if (this.timestamp_max){
//      command += ` --timestamp-max ` + timestamp_max.toString()    }


//      if (this.hits){
//       command += ` --hits ` + hits.toString()    }
      

//       if (this.locked_out){
//         command += ` --locked-out`
//       }
//       command += ` > /dev/null 2>&1`;
//       cy.exec(command);
//  }


// }
// export default ThrottleLog;



import ControlPanel from '../ControlPanel'

class ThrottleLog extends ControlPanel {
  constructor() {
    super()
    this.elements({
      'date' : 'a[data-filter-label="date"]',

      'show' : 'a[data-filter-label="show"]',
      'custom_limit' : 'input[name="perpage"]',

      'search' : 'input[name="filter_by_keyword"]',

      'delete_all' : 'button[rel="modal-confirm-all"]',

     'list' : 'section[class="item-wrap log"]',

     'empty' : 'p[class="no-results"]',
     'confirm' : 'input[type="submit"]',


    

    
    })
  }
}
export default ThrottleLog;


